packages/flutter_tools/lib/src/android/deferred_components_setup_validator.dart

```diff
@@ -0,0 +1,330 @@
+          env.logger.printStatus('Previously existing loading units no longer exist:', emphasis: true);
+          for (final LoadingUnit unit in _goldenComparisonResults['missing'] as Set<LoadingUnit>) {
+            env.logger.printStatus(unit.toString(), color: TerminalColor.grey, indent: 2);
+          }
+          env.logger.printStatus('');
+        }
+        if (_goldenComparisonResults['match'] as bool) {
+          env.logger.printStatus('No change in generated loading units.\n');
+        } else {
+          env.logger.printStatus('''
+It is recommended to verify that the changed loading units are expected
+and to update the `deferred-components` section in `pubspec.yaml` to
+incorporate any changes. The full list of generated loading units can be
+referenced in the $kDeferredComponentsGoldenFileName file located alongside
+pubspec.yaml.
+
+This loading unit check will not fail again on the next build attempt
+if no additional changes to the loading units are detected.
+$_thinDivider\n''');
+        }
+      }
+      // TODO(garyq): Add link to web tutorial/guide once it is written.
+      env.logger.printStatus('''
+Setup verification can be skipped by passing the `--no-verify-deferred-components`
+flag, however, doing so may put your app at risk of not functioning even if the
+build is successful.
+$_thickDivider''');
+      return;
+    }
+    env.logger.printStatus('$_title passed.');
+  }
+
+  void attemptToolExit() {
+    if (exitOnFail && changesNeeded) {
+      throwToolExit('Setup for deferred components incomplete. See recommended actions.', exitCode: 1);
+    }
+  }
+}
+
+// Handles a single deferred component's android dynamic feature module
+// directory.
+class _DeferredComponentAndroidFiles {
+  _DeferredComponentAndroidFiles({
+    @required this.name,
+    @required this.env,
+    Directory templatesDir,
+  }) : _templatesDir = templatesDir;
+
+  // The name of the deferred component.
+  final String name;
+  final Environment env;
+  final Directory _templatesDir;
+
+  Directory get androidDir => env.projectDir.childDirectory('android');
+  Directory get componentDir => androidDir.childDirectory(name);
+
+  File get androidManifestFile => componentDir.childDirectory('src').childDirectory('main').childFile('AndroidManifest.xml');
+  File get buildGradleFile => componentDir.childFile('build.gradle');
+
+  // True when AndroidManifest.xml and build.gradle exist for the android dynamic feature.
+  bool verifyFilesExist() {
+    return androidManifestFile.existsSync() && buildGradleFile.existsSync();
+  }
+
+  // Generates any missing basic files for the dynamic feature into a temporary directory.
+  Future<Map<String, List<File>>> generateFiles({Directory alternateAndroidDir, bool clearAlternateOutputDir = false}) async {
+    final Directory outputDir = alternateAndroidDir == null ? componentDir : alternateAndroidDir.childDirectory(name);
```

Comment:
Maybe something like this to avoid a ternary.

`final Directory outputDir = alternateAndroidDir?.childDirectory(name) ?? componentDir `